src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1064,12 +1049,11 @@ void SolverInterfaceImpl::writeBlockVectorData(
 {
   PRECICE_TRACE(dataID, size);
   PRECICE_CHECK(_state != State::Finalized, "writeBlockVectorData(...) cannot be called after finalize().");
-  PRECICE_VALIDATE_DATA_ID(dataID);
-  if (size == 0)
-    return;
+  PRECICE_REQUIRE_DATA_WRITE(dataID);
   PRECICE_ASSERT(valueIndices != nullptr);
   PRECICE_ASSERT(values != nullptr);
-  PRECICE_REQUIRE_DATA_WRITE(dataID);
+  if (size == 0)
+    return;
```

Comment:
As far as I understand we should rather use `PRECICE_CHECK` here as these quantities are user defined and therefore user variables. We should also add a descriptive error message.
I'm not sure if moving the early return further down: in case the size is indeed zero, i.e., a rank doesn't need to write, `values` and `valueIndices` could be `nullptr` without doing something wrong.

Reply:
That's a good point.
Will make this consistent with the rest as I am already changing parts of these functions.